src/shared/tests.c

```diff
@@ -301,3 +301,26 @@ int enter_cgroup_subroot(char **ret_cgroup) {
 int enter_cgroup_root(char **ret_cgroup) {
         return enter_cgroup(ret_cgroup, false);
 }
+
+const char *ci_environment(void) {
+        /* We return a string because we might want to provide multiple bits of information later on: not
+         * just the general CI environment type, but also whether we're sanitizing or not, etc. The caller is
+         * expected to use strstr on the returned value. */
```

Comment:
why strstr() and not streq()?

Reply:
As the comment says... E.g. we could have "centos", "centos:address-sanitizer", "centos:arch", "centos:fedora:memory-sanitizer", etc. This not supposed to be a general purpose API, but just something to propagate a few bits of information internally. The check with strstr is simple to write.